lib/plugins/ramdisk.js

```diff
@@ -0,0 +1,61 @@
+/*
+  Copyright © 2018 Andrew Powell
+
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+  The above copyright notice and this permission notice shall be
+  included in all copies or substantial portions of this Source Code Form.
+*/
+/* eslint-disable no-param-reassign */
+const crypto = require('crypto');
+const { existsSync, symlinkSync, unlinkSync } = require('fs');
+const { sep } = require('path');
+
+const readPkgUp = require('read-pkg-up');
+const { WebpackPluginRamdisk } = require('webpack-plugin-ramdisk');
+
+const { PluginExistsError } = require('../errors');
+
+module.exports = {
+  init(compiler) {
+    const hasPlugin = compiler.options.plugins.some(
+      (plugin) => plugin instanceof WebpackPluginRamdisk
+    );
+
+    /* istanbul ignore else */
+    if (hasPlugin) {
+      this.log.error(
+        'webpack-plugin-serve adds WebpackRamdiskPlugin automatically. Please remove it from your config.'
+      );
+      throw new PluginExistsError('WebpackRamdiskPlugin exists in the specified configuration.');
+    }
+
+    const pkg = readPkgUp.sync() || {};
+    const { path } = compiler.options.output;
+    const lastSegment = path.split(sep).slice(-1);
```

Comment:
I think `path.basename` is better and safer. This line will fail if `path` ends with `sep`, e.g. `/foo/bar/'

Reply:
that's a great observation. I'll use that instead.